aci/resource_aci_snmpuserp.go

```diff
@@ -0,0 +1,273 @@
+	snmpUserPMap, err := snmpUserP.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != snmpUserP.DistinguishedName {
+		d.Set("snmp_policy_dn", "")
+	} else {
+		d.Set("snmp_policy_dn", GetParentDn(snmpUserP.DistinguishedName, fmt.Sprintf("/"+models.RnSnmpUserP, snmpUserPMap["name"])))
+	}
+	d.Set("annotation", snmpUserPMap["annotation"])
+	authKey := d.Get("authorization_key").(string)
+	if authKey != "" {
+		d.Set("authorization_key", snmpUserPMap["authKey"])
+	}
+	d.Set("authorization_type", snmpUserPMap["authType"])
+	d.Set("name", snmpUserPMap["name"])
+	d.Set("name_alias", snmpUserPMap["nameAlias"])
+	privKey := d.Get("privacy_key").(string)
+	if privKey != "" {
+		d.Set("privacy_key", snmpUserPMap["privKey"])
+	}
+	d.Set("privacy_type", snmpUserPMap["privType"])
+	return d, nil
+}
+
+func resourceAciSnmpUserProfileImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	snmpUserP, err := getRemoteUserProfile(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setUserProfileAttributes(snmpUserP, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciSnmpUserProfileCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] SNMP User Profile: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	name := d.Get("name").(string)
+	SNMPPolicyDn := d.Get("snmp_policy_dn").(string)
+
+	snmpUserPAttr := models.SnmpUserProfileAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		snmpUserPAttr.Annotation = Annotation.(string)
+	} else {
+		snmpUserPAttr.Annotation = "{}"
+	}
+
+	if AuthKey, ok := d.GetOk("authorization_key"); ok {
+		snmpUserPAttr.AuthKey = AuthKey.(string)
+	}
+
+	if AuthType, ok := d.GetOk("authorization_type"); ok {
+		snmpUserPAttr.AuthType = AuthType.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		snmpUserPAttr.Name = Name.(string)
+	}
+
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		snmpUserPAttr.NameAlias = NameAlias.(string)
+	}
+
+	if PrivKey, ok := d.GetOk("privacy_key"); ok {
+		snmpUserPAttr.PrivKey = PrivKey.(string)
+	}
+
+	if PrivType, ok := d.GetOk("privacy_type"); ok {
+		snmpUserPAttr.PrivType = PrivType.(string)
+	}
+	snmpUserP := models.NewSnmpUserProfile(fmt.Sprintf(models.RnSnmpUserP, name), SNMPPolicyDn, desc, snmpUserPAttr)
+
+	err := aciClient.Save(snmpUserP)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(snmpUserP.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciSnmpUserProfileRead(ctx, d, m)
+}
+func resourceAciSnmpUserProfileUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
If we do not consider the annotation, description and name_alias attributes. We can remove the update function.

Reply:
Lets keep the update function as it is needed for those 3.